src/sphere.jl

```diff
@@ -14,5 +14,5 @@ Base.:(==)(s1::Sphere, s2::Sphere) = s1.c==s2.c && s1.r==s2.r && s1.data==s2.dat
 Base.hash(s::Sphere, h::UInt) = hash(s.c, hash(s.r, hash(s.data, hash(:Sphere, h))))
 
 Base.in(x::SVector{N}, s::Sphere{N}) where {N} = sum(abs2,x - s.c) ≤ s.r^2
-normal(x::SVector{N}, s::Sphere{N}) where {N} = normalize(x - s.c)
+surfpt_nearby(x::SVector{N}, s::Sphere{N}) where {N} = (nout = normalize(x - s.c); (s.c+s.r*nout, nout))
```

Comment:
fails if `x == s.c`...

Another possibility in these cases is to return `nout == zero vector`, and have the subpixel averaging code check for that case.  (Zero normal vector = normal couldn't be determined, and it should abandon subpixel averaging for that point.)